tests/migrations/namespace-form-fields.js

```diff
@@ -0,0 +1,147 @@
+      },
+      "patient_name": {
+        "labels": {
+          "tiny": {
+            "en": "N"
+          },
+          "description": {
+            "en": "Patient Name"
+          },
+          "short": {
+            "en": "Name"
+          }
+        },
+        "type": "string"
+      }
+    },
+    "public_form": true,
+    "use_sentinel": true
+  }
+};
+
+exports['run does nothing if no data records'] = function(test) {
+  test.expect(2);
+  sinon.stub(config, 'load').callsArg(0);
+  var getView = sinon.stub(db.medic, 'view').callsArgWith(3, null, { rows: [] });
+  var getConfig = sinon.stub(config, 'get').returns({});
+  migration.run(function(err) {
+    test.equals(err, undefined);
+    test.equals(getView.callCount, 1);
+    test.done();
+  });
+};
+
+exports['run does nothing if report already migrated'] = function(test) {
+  test.expect(3);
+  var doc = {
+    _id: 'a',
+    reported_date: '123',
+    form: 'P',
+    fields: { name: 'michael' }
+  };
+  sinon.stub(config, 'load').callsArg(0);
+  var getView = sinon.stub(db.medic, 'view').callsArgWith(3, null, { rows: [ { id: 'a' } ] });
+  var getDoc = sinon.stub(db.medic, 'get').callsArgWith(1, null, doc);
+  var getConfig = sinon.stub(config, 'get').returns({});
+  migration.run(function(err) {
+    test.equals(err, undefined);
+    test.equals(getView.callCount, 1);
+    test.equals(getDoc.callCount, 1);
+    test.done();
+  });
+};
+
+exports['run does nothing if no form'] = function(test) {
+  test.expect(3);
+  var doc = {
+    _id: 'a',
+    reported_date: '123'
+  };
+  sinon.stub(config, 'load').callsArg(0);
+  var getView = sinon.stub(db.medic, 'view').callsArgWith(3, null, { rows: [ { id: 'a' } ] });
+  var getDoc = sinon.stub(db.medic, 'get').callsArgWith(1, null, doc);
+  var getConfig = sinon.stub(config, 'get').returns({});
+  migration.run(function(err) {
+    test.equals(err, undefined);
+    test.equals(getView.callCount, 1);
+    test.equals(getDoc.callCount, 1);
+    test.done();
+  });
+};
+
+exports['run migrates report'] = function(test) {
+  test.expect(8);
+  var doc = {
+    _id: 'a',
+    reported_date: '123',
+    form: 'P',
+    last_menstrual_period: 22,
+    patient_name: 'sarah'
+  };
+  var expected = {
+    last_menstrual_period: 22,
+    patient_name: 'sarah'
+  };
+  sinon.stub(config, 'load').callsArg(0);
+  var getView = sinon.stub(db.medic, 'view').callsArgWith(3, null, { rows: [ { id: 'a' } ] });
+  var getDoc = sinon.stub(db.medic, 'get').callsArgWith(1, null, doc);
+  var getConfig = sinon.stub(config, 'get').returns(forms);
+  var insert = sinon.stub(db.medic, 'insert').callsArgWith(1, null, null);
+  migration.run(function(err) {
+    test.equals(err, undefined);
+    test.equals(getView.callCount, 1);
+    test.equals(getDoc.callCount, 1);
+    test.equals(getConfig.callCount, 1);
+    test.equals(insert.callCount, 1);
+    test.same(insert.args[0][0].fields, expected);
+    test.equals(insert.args[0][0].last_menstrual_period, null);
```

Comment:
If we don't want them on the doc then we should probably assert they are undefined?